tools/ld-dropout-correct/correctorpool.cpp

```diff
@@ -166,29 +165,46 @@ bool CorrectorPool::getInputFrame(qint32& frameNumber,
     secondFieldVideoData.resize(numberOfSources);
     secondFieldMetadata.resize(numberOfSources);
     videoParameters.resize(numberOfSources);
+    sourceFrameQuality.resize(numberOfSources);
 
     // Get the current VBI frame number based on the first source
-    qint32 currentVbiFrame = convertSequentialFrameNumberToVbi(frameNumber, 0);
+    qint32 currentVbiFrame = -1;
+    if (numberOfSources > 1) currentVbiFrame = convertSequentialFrameNumberToVbi(frameNumber, 0);
     for (qint32 sourceNo = 0; sourceNo < numberOfSources; sourceNo++) {
         // Determine the fields for the input frame
         if (sourceNo == 0) {
             // No need to perform VBI frame number mapping on the first source
-            firstFieldNumber[sourceNo] = ldDecodeMetaData[0].getFirstFieldNumber(frameNumber);
-            secondFieldNumber[sourceNo] = ldDecodeMetaData[0].getSecondFieldNumber(frameNumber);
+            firstFieldNumber[sourceNo] = ldDecodeMetaData[sourceNo]->getFirstFieldNumber(frameNumber);
```

Comment:
This looks the same as the block below.

Restructure as if (sourceNo != 0 && currentVbiFrame >= ...) { set to -1; } else { this block of code }?

Reply:
They are not quite the same, but I restructured it a bit to make it clearer